src/shared/copy.c

```diff
@@ -346,8 +346,12 @@ int copy_bytes_full(
 
                         /* Make sure we're not copying more than the current data segment. */
                         m = MIN(m, (size_t) e - c);
+                        if (m <= 0)
+                                continue;
                 }
 
+                assert(m > 0);
```

Comment:
Adding an assert in a core library at this stage is too risky, if you want to keep it we can postpone this to 258, otherwise please drop it

Reply:
then we should postpone to 258, adding a crash in a core area moments before a release is not safe

Reply:
I totally agree with @YHNdnzj, but I do not want to delay v257 more.
I will soon re-add the assertion after v257.